### PR TITLE
Implementation HS256 signature verification

### PR DESCRIPTION
**Pull request by damieng**
Implements (but does not yet wire up) the `HS256` signature verification.

### CODE REVIEW

File: src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs [hunk 1]
```diff
@@ -33,9 +33,8 @@ internal static async Task AssertTokenMeetsRequirements(this IdTokenRequirements
 
             var token = DecodeToken(rawIDToken);
 
-            // For now we want to support HS256 + ClientSecret as we just had a major release.
-            // TODO: In the next major (v4.0) we should remove this condition as well as Auth0ClientOptions.ClientSecret
```
Comment (lbalmaceda) [inline src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs]: I'm confused with this comment about v4 and the one below for v7. Should we keep supporting client_secret (and hs256) or not?

File: src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs [hunk 2]
```diff
@@ -33,9 +33,8 @@ internal static async Task AssertTokenMeetsRequirements(this IdTokenRequirements
...
-            if (token.SignatureAlgorithm != "HS256")
+            // TODO: Make signatureVerifier non-optional and not skipped for 'HS256' in 7.0.0
+            if (token.SignatureAlgorithm != "HS256" || signatureVerifier != null)
```
Comment (lbalmaceda) [inline src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs]: where is `signatureVerifier` set and on which scenarios? and more importantly what if the token is signed with `alg=none`?

File: src/Auth0.AuthenticationApi/Tokens/SymmetricSignatureVerifier.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    internal class SymmetricSignatureVerifier : ISignatureVerifier
+    {
+        private readonly SecurityKey signingKey;
+
+        public SymmetricSignatureVerifier(SecurityKey signingKey)
+        {
+            this.signingKey = signingKey;
+        }
+
+        public static SymmetricSignatureVerifier FromClientSecret(string clientSecret, bool isBase64Encoded = false)
```
Comment (lbalmaceda) [inline src/Auth0.AuthenticationApi/Tokens/SymmetricSignatureVerifier.cs]: `isBase64Encoded ` is legacy (+3 years old setting). Please remove if this hits a new major.

File: src/Auth0.AuthenticationApi/Tokens/SymmetricSignatureVerifier.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                    ValidateIssuerSigningKey = true,
+                    IssuerSigningKey = signingKey
+                };
+
+                securityTokenHandler.ValidateToken(token, validationParameters, out var verifiedToken);
+                return verifiedToken;
+            }
+            catch (SecurityTokenException e)
+            {
+                throw new IdTokenValidationException("Invalid ID token signature.", e);
```
Comment (lbalmaceda) [inline src/Auth0.AuthenticationApi/Tokens/SymmetricSignatureVerifier.cs]: According to the docs, the error message should be

```suggestion
                throw new IdTokenValidationException("Invalid token signature.", e);
```

File: tests/Auth0.AuthenticationApi.IntegrationTests/Tokens/IdTokenValidatorTests.cs [hunk 5]
```diff
@@ -61,31 +61,45 @@ public async void ThrowsWhenIdTokenCanNotBeDecoded()
         }
 
         [Fact]
```
Comment (lbalmaceda) [inline tests/Auth0.AuthenticationApi.IntegrationTests/Tokens/IdTokenValidatorTests.cs]: Please add a test to cover the `none` algorithm case

File: src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs (hunk 1 above)
Comment (damieng) [reply src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs]: This was a hold-over from borrowing the code from auth0-oidc-client-net and can be ignored.

File: tests/Auth0.AuthenticationApi.IntegrationTests/Tokens/IdTokenValidatorTests.cs (hunk 5 above)
Comment (damieng) [reply tests/Auth0.AuthenticationApi.IntegrationTests/Tokens/IdTokenValidatorTests.cs]: That test already exists here https://github.com/auth0/auth0.net/blob/master/tests/Auth0.AuthenticationApi.IntegrationTests/Tokens/IdTokenValidatorTests.cs#L93

File: src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs (hunk 2 above)
Comment (damieng) [reply src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs]: That will be in the next PR that actually allows the HS256 validation to be used.
